### PR TITLE
SVG templates

### PR DESCRIPTION
**Pull request by jmleroux**
Extract SVG templates from renderer classes.
Easier to maintain, easier to validate, easier to test.

### CODE REVIEW

**Comment by jmleroux** [comment]
@liuggio ,
a little bit of refactoring. What do you think of it ?

I extracted all SVG in externals files.
I added the abstract LocalSvgRenderer to extract duplicate code.

**Comment by jmleroux** [comment]
Nobody ?

File: spec/PUGX/Poser/Render/SvgRenderSpec.php [hunk 1]
```diff
@@ -24,12 +24,12 @@ function it_should_render_a_svg()
     public function getMatchers()
     {
         return array(
-            'beAValidSVGImage' => function($subject) {
+            'beAValidSVGImage' => function ($subject) {
```
Comment (stof) [inline spec/PUGX/Poser/Render/SvgRenderSpec.php]: This matcher may be improved by checking that it is a valid XML

File: src/Render/LocalSvgRenderer.php [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+ * @author JM Leroux <jmleroux.pro@gmail.com>
+ */
+abstract class LocalSvgRenderer implements RenderInterface
+{
+    const VENDOR_COLOR = '#555';
+
+    /**
+     * @var TextSizeCalculatorInterface
+     */
+    protected $textSizeCalculator;
```
Comment (stof) [inline src/Render/LocalSvgRenderer.php]: I think using private visibility would be better

File: src/Render/SvgFlatRender.php [hunk 3]
```diff
@@ -11,130 +11,14 @@
...
-        $parameters['vendorColor']         = self::VENDOR_COLOR;
-        $parameters['valueColor']          = $badge->getHexColor();
-        $parameters['vendor']              = $badge->getSubject();
-        $parameters['value']               = $badge->getStatus();
-        $parameters['vendorStartPosition'] = round($parameters['vendorWidth'] / 2, 1) + 1;
-        $parameters['valueStartPosition']  = $parameters['vendorWidth'] + round($parameters['valueWidth'] / 2, 1) - 1;
-
-        return $this->renderSvg(self::$template, $parameters, $badge->getFormat());
-    }
+    protected $templateName = 'flat';
```
Comment (stof) [inline src/Render/SvgFlatRender.php]: expecting child class to overwrite a protected property to initialize it looks weird to me

File: src/Render/LocalSvgRenderer.php [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+
+        return Image::createFromString($render, $format);
+    }
+
+    /**
+     * @param Badge $badge
+     *
+     * @return array
+     */
+    protected function buildParameters(Badge $badge)
```
Comment (stof) [inline src/Render/LocalSvgRenderer.php]: This should stay private

File: spec/PUGX/Poser/Render/SvgRenderSpec.php (hunk 1 above)
Comment (jmleroux) [inline spec/PUGX/Poser/Render/SvgRenderSpec.php]: I just added a very basic xml validation step.

File: src/Render/LocalSvgRenderer.php [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+    }
+
+    /**
+     * @param string $render
+     * @param array  $parameters
+     * @param string $format
+     *
+     * @return Image
+     */
+    protected function renderSvg($render, $parameters, $format)
```
Comment (stof) [inline src/Render/LocalSvgRenderer.php]: these methods should also be private

**Comment by jmleroux** [comment]
You can also have a look at the Scrutinizer results : https://scrutinizer-ci.com/g/jmleroux/poser/inspections/52944522-d361-4bdd-bea3-51bcd2f037e9

**Comment by jmleroux** [comment]
anybody to gia GTM ?
@liuggio , @stof or other ?

**Comment by liuggio** [comment]
@jmleroux great job!!
adding new release soon
thanks @stof 
and @toretto460

**Pull request merged**